tests/unit/cn/cndb_test.c

```diff
@@ -755,22 +759,51 @@ MTF_DEFINE_UTEST_PREPOST(cndb_test, cndb_blkdel_test, test_pre, test_post)
         TAG_3 = 3,
     };
 
-    struct cndb_txc  txc = { { CNDB_TYPE_TXC }, CNID_1, TXID_2, TAG_3, 0, 0, 0 };
-    struct cndb_txd  txd = { { CNDB_TYPE_TXD }, CNID_1, TXID_2, TAG_3, 0 };
+    err = mpm_mblock_alloc(MPOOL_MBLOCK_SIZE_DEFAULT, &hoidc);
+    ASSERT_EQ(0, merr_errno(err));
+
+    err = mpm_mblock_alloc(MPOOL_MBLOCK_SIZE_DEFAULT, &hoidd);
+    ASSERT_EQ(0, merr_errno(err));
+
+    union cndb_mtu txc = {
+        .c = {
+            .hdr = { CNDB_TYPE_TXC },
+            .mtc_cnid = CNID_1,
+            .mtc_id = TXID_2,
+            .mtc_tag = TAG_3,
+            .mtc_keepvbc = 0,
+            .mtc_hoid = hoidc,
+            .mtc_kcnt = 0,
+            .mtc_vcnt = 0,
+        },
+    };
+
+    union cndb_mtu txd = {
```

Comment:
Made these guys unions because I was getting a stack overflow buffer error from asan. For whatever reason this fixed it.

Reply:
Weird.  Maybe some function that accepts a cndb_txc thinks it's actually a cndb_mtu and does bad things?

Reply:
Hmm, looks like cndb_blkdel() wants a cndb_mtu, but we were passing it a cndb_txc previously...

Reply:
cndb_mtu is a union including cndb_txc, so it should just work I thought if you passed the address of a cndb_txc.